### PR TITLE
Alter vcrpy config to ignore the "id" in each request

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Update the test config for vcrpy so that it doesn't take the 'id' into account.

I've also changed the setup so that vcrpy is accessed via a plugin, `pytest-recording`, which makes it a bit simpler to add recordings to tests. A slightly unfortunate side effect is that all the cassette files have been renamed and moved to a different directory.

ETA: codecov upload failed

### CODE REVIEW

File: src/biokbase/narrative/tests/conftest.py [hunk 1]
```diff
@@ -1,15 +1,84 @@
...
+
+def body_matcher(r1: vcr.request.Request, r2: vcr.request.Request) -> None:
+    """Compare the body contents of two requests to work out if they are identical or not."""
+    r1_body = json.loads(r1.body.decode())
+    r2_body = json.loads(r2.body.decode())
+    if "id" in r1_body:
+        del r1_body["id"]
+    if "id" in r2_body:
+        del r2_body["id"]
+    assert r1_body == r2_body
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/conftest.py:52]: that pesky 'id' param makes it more difficult for vcrpy to match requests, so this removes it and then checks to see if the requests are identical or not.

File: src/biokbase/narrative/tests/generate_test_results.py [hunk 2]
```diff
@@ -43,13 +44,13 @@
     TEST_SPECS[tag] = spec_dict
 
 
-def get_test_spec(tag: str, app_id: str) -> dict[str, dict]:
+def get_test_spec(tag: str, app_id: str) -> dict[str, dict[str, Any]]:
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/generate_test_results.py:47]: fix up the bare `dict` annotations so the dict contents are typed

File: src/requirements-dev.txt [hunk 3]
```diff
@@ -2,7 +2,8 @@ black==24.1.1
 coverage==7.4.1
 flake8==7.0.0
 isort==5.13.2
-pytest==8.0.0
+pytest==8.2.0
```
Comment (ialarmedalien) [inline src/requirements-dev.txt:5]: next PR will be other python updates so I'll put the version upgrade in that one

File: src/requirements-dev.txt [hunk 4]
```diff
@@ -2,7 +2,8 @@ black==24.1.1
...
 pytest-cov==4.1.0
+pytest-recording==0.13.1
```
Comment (ialarmedalien) [inline src/requirements-dev.txt:7]: new plugin for running vcrpy in pytest

File: src/biokbase/narrative/tests/test_app_util.py [hunk 5]
```diff
@@ -368,8 +366,9 @@ def test_transform_param_value_textsubdata(value, expected):
 
 @pytest.mark.parametrize("tf_type", all_types)
 @pytest.mark.parametrize("inp", input_list)
+@pytest.mark.vcr
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_app_util.py:369]: use vcrpy to record responses

File: src/biokbase/narrative/tests/conftest.py [hunk 6]
```diff
@@ -1,15 +1,71 @@
...
+@pytest.fixture(scope="session")
+def user_name(test_config: ConfigTests) -> str:
+    return test_config.get("users", "test_user")
+
+
+@pytest.fixture(scope="session")
+def user_token(test_config: ConfigTests) -> str | None:
+    return read_token_file(
+        test_config.get_path("token_files", "test_user", from_root=True)
+    )
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/conftest.py:33]: created a couple of fixtures in the pytest style.

File: src/biokbase/narrative/tests/cassettes/test_app_util/test_transform_param_value_all_types_with_all_inputs[name-list-putative-ref-].yaml [hunk 7]
```diff
@@ -2,7 +2,7 @@ interactions:
 - request:
     body: '{"method": "Workspace.get_object_info3", "params": [{"objects": [{"ref":
       "test_narrative:narrative_1695311690413/Abiotrophia_defectiva_ATCC_49176"}]}],
-      "version": "1.1", "id": "2675810953611064"}'
+      "version": "1.1", "id": "6197470998671907"}'
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/cassettes/test_app_util/test_transform_param_value_all_types_with_all_inputs[name-list-putative-ref-].yaml:5]: Lots of cassettes have been moved and in some cases, renamed.

File: scripts/narrative_backend_tests.sh [hunk 8]
```diff
@@ -3,12 +3,6 @@ export NARRATIVE_DIR=$(pwd)
 
 docker_container=${DOCKER_CONTAINER:-false}
 
-# if the test is being run in a docker container, ensure
-# that the conda base environment is activated
-if [ "$docker_container" == "true" ]; then
-    source activate base
```
Comment (ialarmedalien) [inline scripts/narrative_backend_tests.sh:9]: this line is no longer relevant

File: src/biokbase/narrative/tests/test_app_util.py [hunk 9]
```diff
@@ -368,8 +364,9 @@ def test_transform_param_value_textsubdata(value, expected):
 
 @pytest.mark.parametrize("tf_type", all_types)
 @pytest.mark.parametrize("inp", input_list)
+@pytest.mark.vcr()
```
Comment (briehl) [inline src/biokbase/narrative/tests/test_app_util.py:367]: Much nicer! 🎉

**Review by briehl — APPROVED** [review]
Looks like all good changes. I like the pytest vcr plugin!

**Pull request merged**